### PR TITLE
added set_kth_bit.cpp

### PR DESCRIPTION
**Pull request by aman2000raj**
#### Description of Change

Given a number N and a value K. From the right, set the Kth bit in the binary representation of N. The position of Least Significant Bit(or last bit) is 0, the second last bit is 1 and so on.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:  added an algorithm which set the kth bit from right of a given Number.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1859"></a>

### CODE REVIEW

File: bit_manipulation/set_kth_bit.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+
```
Comment (Panquesito7) [inline bit_manipulation/set_kth_bit.cpp:1]:
```suggestion
```

File: bit_manipulation/set_kth_bit.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ */
+namespace bit_manipulation {
+/**
+ * @namespace setKthBit
+ * @brief Functions for the [From the right, set the Kth bit in the binary
+ * representation of N]
+ * (https://practice.geeksforgeeks.org/problems/set-kth-bit3724/1/)
+ * implementation
+ */
+namespace Set_kth_bit {
```
Comment (Panquesito7) [inline bit_manipulation/set_kth_bit.cpp:35]:
```suggestion
namespace set_kth_bit {
```

File: bit_manipulation/set_kth_bit.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    int pos = 1 << k;  // "pos" variable is used to store 1 at kth postion and
+                      // rest bits are 0. in binary representation of number 'n'
+
+    return N | pos;  // by taking or with the pos and the N we set the bit of N
+                     // at kth position.
+}
+}  // namespace Set_kth_bit
+}  // namespace bit_manipulation
+
+static void test() {
```
Comment (Panquesito7) [inline bit_manipulation/set_kth_bit.cpp:55]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test() {
```

File: bit_manipulation/set_kth_bit.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    std ::int64_t k) {  // int64_t is preferred over int so that
+                        // no Overflow can be there.
+
+    int pos = 1 << k;  // "pos" variable is used to store 1 at kth postion and
+                      // rest bits are 0. in binary representation of number 'n'
+
+    return N | pos;  // by taking or with the pos and the N we set the bit of N
+                     // at kth position.
+}
+}  // namespace Set_kth_bit
```
Comment (Panquesito7) [inline bit_manipulation/set_kth_bit.cpp:52]:
```suggestion
}  // namespace set_kth_bit
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 😄👍
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/aman2000raj/C-Plus-Plus/actions

**Review by Qannaf — APPROVED** [review]

**Pull request closed without merging**